projects/superfluid-automated-vesting/src/test/VestingAutomation.t.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.16;
+
+import { ISuperToken } from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol";
+import { FlowOperatorDefinitions } from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import { SuperfluidFrameworkDeployer, SuperfluidTester, Superfluid, ConstantFlowAgreementV1, CFAv1Library } from "./SuperfluidTester.t.sol";
+import { ERC1820RegistryCompiled } from "@superfluid-finance/ethereum-contracts/contracts/libs/ERC1820RegistryCompiled.sol";
+ import {AutomateTaskCreator} from '../gelato/AutomateTaskCreator.sol';
+import { VestingAutomation } from "../VestingAutomation.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+ import { VestingScheduler } from "../VestingScheduler.sol";
+import { IVestingScheduler } from "../interface/IVestingScheduler.sol";
+/// @title VestingAutomationTests
+contract VestingAutomationTests is SuperfluidTester {
+
+  
+
+    /// @dev This is required by solidity for using the CFAv1Library in the tester
+    using CFAv1Library for CFAv1Library.InitData;
+
+    SuperfluidFrameworkDeployer internal immutable sfDeployer;
+    SuperfluidFrameworkDeployer.Framework internal sf;
+    Superfluid host;
+    ConstantFlowAgreementV1 cfa;
+    VestingAutomation internal _vestingAutomation;
+    VestingScheduler internal _vestingScheduler;
+    uint256 private _expectedTotalSupply = 0;
+    CFAv1Library.InitData internal cfaV1Lib; 
+    ERC20 private _token;
+    ISuperToken private _superToken;
+
+    /// @dev Constants for Testing
+     address private constant _automate = address(0);
+      address private constant _primaryFundsOwner = address(1);
+    
+    uint32 immutable START_DATE = uint32(block.timestamp + 1);
+    uint32 immutable CLIFF_DATE = uint32(block.timestamp + 10 days);
+    int96 constant FLOW_RATE = 1000000000;
+    uint256 constant CLIFF_TRANSFER_AMOUNT = 1 ether;
+    uint32 immutable END_DATE = uint32(block.timestamp + 20 days);
+    bytes constant EMPTY_CTX = "";
+
+    constructor() SuperfluidTester(3) {
+        vm.startPrank(admin);
+        vm.etch(ERC1820RegistryCompiled.at, ERC1820RegistryCompiled.bin);
+        sfDeployer = new SuperfluidFrameworkDeployer();
+        sf = sfDeployer.getFramework();
+        host = sf.host;
+        cfa = sf.cfa;
+   
+       // vm.stopPrank();
+        // can be an empty string in dev or testnet deployments
+        string memory registrationKey = "";
+          _vestingScheduler = new VestingScheduler(host, "");
+      
+
+        cfaV1Lib = CFAv1Library.InitData(host,cfa);
+             vm.stopPrank();
+               _vestingAutomation = new VestingAutomation( _superToken,  _automate,  _primaryFundsOwner, _vestingScheduler);
```

Comment:
this part error 
├─ [811] SuperTokenFactory::castrate()
    │   │   │   └─ ← "Initializable: contract is already initialized"

─ [2615] VestingScheduler::createVestingSchedule(UUPSProxy: [0x6a90a70c382A33cf88e7672F6a9509aD2D2a7663], 0x0000000000000000000000000000000000000422, 2, 864001, 1000000000, 1000000000000000000, 1728001, 0x)
    │   └─ ← "ScheduleAlreadyExists()